### PR TITLE
[Storage] Boundary for service version.

### PR DESCRIPTION
**Pull request by kasobol-msft**
Create annotation for declaring service version boundary.

Fixed bug in datalake where service version was not mapped correctly into blob version.

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlockBlobAPITest.groovy [hunk 1]
```diff
@@ -1949,6 +1951,7 @@ class BlockBlobAPITest extends APISpec {
         response.getHeaders().getValue("x-ms-version") == "2017-11-09"
     }
 
+    @RequiredServiceVersion(clazz = BlobServiceVersion.class, min = "V2020_04_08")
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlockBlobAPITest.groovy:1954]: Ideally we'd package new features in separate test class going forward so we don't have this verbosity.

File: sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/RequiredServiceVersion.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.TYPE, ElementType.METHOD})
+@ExtensionAnnotation(RequiredServiceVersionExtension.class)
+public @interface RequiredServiceVersion {
+    Class<? extends Enum<?>> clazz();
+    String min();
+}
```
Comment (kasobol-msft) [inline sdk/storage/azure-storage-common/src/test-shared/java/com/azure/storage/common/test/shared/extensions/RequiredServiceVersion.java:19]: This isn't the prettiest. These are boundaries:
- java annotation has quite restrictive list of types that can be used to declare it's values.
- We don't want to repeat this code and accompanying extension across modules. (to make it worse nio and batch would need same copy as blobs would have..)

I'm open to suggestions here.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**